lib/src/main/java/jpprof/CPUProfiler.java

```diff
@@ -56,10 +56,30 @@ public static void start(Duration duration, OutputStream out) throws IOException
      * a tmp path, and returns the absolute path.
      */
     private static String copyLibrary() throws Exception {
-        String osName = System.getProperty("os.name").toLowerCase();
-        String osArch = System.getProperty("os.arch").toLowerCase();
 
         String embeddedLibPrefix = "/async-profiler-libs/libasyncProfiler";
+        String embeddedLibSuffix = getLibrarySuffix();
+
+        InputStream is = CPUProfiler.class.getResourceAsStream(embeddedLibPrefix + embeddedLibSuffix);
+        if (is == null) {
+            return System.getProperty("user.dir") + "/async-profiler-2.8.3/" + "libasyncProfiler"
```

Comment:
Something for a followup: the path incorporates the async-profiler version currently, we should either remove that from the path or find a way to keep things automatically up-to-date.

Reply:
Yes I wasn't sure yet.